calculator.rb

```diff
@@ -0,0 +1,90 @@
+# prints a welcome message to the user
+print "Welcome to the calculating program! which operater you want to use \n \
+1. add(+)\n 2. substract(-)\n 3. multiply(*)\n 4. divide(/)\n"
+
+# prompt user to enter a math operater
+print "Please enter a math operater (name or a symbol): "
+math_operater = gets.chomp.downcase
+
+# creates an array with list of maths operator
+math_operators = ["+", "-", "/", "*", "add", "subtract", "multiply", "divide"]
+
+# ask user to input an maths operater until its valid
+until math_operators.include?(math_operater)
+    puts"Invalid math operator! Please try again."
+    print "Please enter a math operater: "
+    math_operater = gets.chomp.downcase
+end
+
+# rescue clause catches the argument error 
+begin
+    print "Please enter the first number: "
+    num_one = Integer(gets.chomp) 
+rescue  ArgumentError  => e
+    puts "You did not enter a a number. Try again!"
+end
+
+until num_one.is_a? Numeric
+   
+    # rescue clause catches the argument error 
+    begin
+        print "Please enter the first number: "
+        num_one = Integer(gets.chomp)
+    rescue  ArgumentError  => e
+        puts "You did not enter a a number. Try again!"
+    end
```

Comment:
We will get into `begin ... rescue` clauses in-depth later in the program! For now, this is an okay solution, but it may be interesting to consider what it would look like without using the `begin ... rescue`